src/engraving/libmscore/note.cpp

```diff
@@ -2261,6 +2262,16 @@ void Note::layout2()
         qreal d  = score()->point(score()->styleS(Sid::dotNoteDistance)) * mag();
         qreal dd = score()->point(score()->styleS(Sid::dotDotDistance)) * mag();
         qreal x  = chord()->dotPosX() - pos().x() - chord()->pos().x();
+        // adjust dot distance for hooks
+        if (chord()->hook() && chord()->up()) {
+            qreal hookRight = chord()->hook()->width() + chord()->hook()->x() + chord()->pos().x();
+            qreal hookBottom = chord()->hook()->height() + chord()->hook()->y() + chord()->pos().y() + (0.25 * spatium());
+            // the top dot in the chord, not the dot for this particular note:
+            qreal dotY = chord()->notes().back()->y() + chord()->notes().back()->dots().first()->pos().y();
+            if (chord()->dotPosX() < hookRight && dotY < hookBottom) {
+                d = chord()->hook()->width() + (0.25 * spatium());
```

Comment:
are both these `0.25` the same number? If so can't we consolidate those into one variable?

Reply:
They are not. The first one is adding a little bit of margin to the bottom of a hook for collision detection purposes (because otherwise they could be still right next to each other and not be technically colliding), and the second adds margin to the right side of the hook. Vertical vs horizontal positioning.

Reply:
Thanks for explaining.